### ISSUE TITLE
Email/import with complex message - non text/* parts have no partId?

### ISSUE THREAD

**Issue by wolfsage** [cyrusimap/cyrus-imapd#2402]
_TLDR: Any non text/* part does not have a partId but the spec seems to indicate it should._

At the top of this branch is a new test:

  https://github.com/wolfsage/JMAP-TestSuite/tree/broken-partid-for-images

With the following email being imported (meant to mirror the example given in the block just above here: https://github.com/jmapio/jmap/blob/master/spec/mail/message.mdown#emailget )

(Note that this is *hand written* so perhaps there's bugs in it??)

````
Date: Thu, 21 Jun 2018 11:00:06 -0400
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Content-Type: multipart/mixed; boundary="TOP"

--TOP
Date: Thu, 21 Jun 2018 11:00:06 -0400
MIME-Version: 1.0
Content-Type: text/plain
Content-Transfer-Encoding: 7bit

This is text part A

--TOP
Date: Thu, 21 Jun 2018 11:00:06 -0400
MIME-Version: 1.0
Content-Type: multipart/mixed; boundary="MIXED1"
Content-Transfer-Encoding: 7bit

--MIXED1
Date: Thu, 21 Jun 2018 11:00:06 -0400
MIME-Version: 1.0
Content-Type: multipart/alternative; boundary="ALTERNATIVE1";
Content-Transfer-Encoding: 7bit

--ALTERNATIVE1
Date: Thu, 21 Jun 2018 11:00:06 -0400
MIME-Version: 1.0
Content-Type: multipart/mixed; boundary="MIXED2";
Content-Transfer-Encoding: 7bit

--MIXED2
Date: Thu, 21 Jun 2018 11:00:06 -0400
MIME-Version: 1.0
Content-Type: text/plain
Content-Disposition: inline
Content-Transfer-Encoding: 7bit

This is text part B

--MIXED2
Date: Thu, 21 Jun 2018 12:00:19 -0400
MIME-Version: 1.0
Content-Type: image/jpeg; name="c.jpg"
Content-Disposition: inline
Content-Transfer-Encoding: quoted-printable

=FF=D8=FF=E0=00=10JFIF=00=01=01=01=00H=00H=00=00=FF=FE=00=13Created with GI=
MP=FF=DB=00C=00=03=02=02=03=02=02=03=03=03=03=04=03=03=04=05=08=05=05=04=04=
=05
=07=07=06=08=0C
=0C=0C=0B
=0B=0B=0D=0E=12=10=0D=0E=11=0E=0B=0B=10=16=10=11=13=14=15=15=15=0C=0F=17=18=
=16=14=18=12=14=15=14=FF=DB=00C=01=03=04=04=05=04=05	=05=05	=14=0D=0B=0D=14=
=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=
=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=FF=
=C2=00=11=08=00
=00
=03=01=11=00=02=11=01=03=11=01=FF=C4=00=15=00=01=01=00=00=00=00=00=00=00=00=
=00=00=00=00=00=00=02=07=FF=C4=00=14=01=01=00=00=00=00=00=00=00=00=00=00=00=
=00=00=00=00=00=FF=DA=00=0C=03=01=00=02=10=03=10=00=00=01=AF=94=A0=0C=FF=C4=
=00=18=10=01=01=00=03=00=00=00=00=00=00=00=00=00=00=00=00=02=05=00=03=10=FF=
=DA=00=08=01=01=00=01=05=02=A3U=EB=B5=88=17=CF=FF=C4=00=14=11=01=00=00=00=
=00=00=00=00=00=00=00=00=00=00=00=00 =FF=DA=00=08=01=03=01=01?=01=1F=FF=C4=
=00=14=11=01=00=00=00=00=00=00=00=00=00=00=00=00=00=00=00 =FF=DA=00=08=01=
=02=01=01?=01=1F=FF=C4=00=1F=10=00=02=01=04=02=03=00=00=00=00=00=00=00=00=
=00=00=01=03=02=05=10=11!=121#q=A1=FF=DA=00=08=01=01=00=06?=02z=1FXM=1Dj=8C=
$=98=B0C=CD=9E=C9=E5=DE=F5=81=8F=B6=D8=07=DD=BF=FF=C4=00=18=10=01=01=01=01=
=01=00=00=00=00=00=00=00=00=00=00=00=01=11!=00=10=FF=DA=00=08=01=01=00=01?!=
=98=C0=8C!=A3`=0D(=95q:=06n=91g=9F=FF=DA=00=0C=03=01=00=02=00=03=00=00=00=
=10=12O=FF=C4=00=14=11=01=00=00=00=00=00=00=00=00=00=00=00=00=00=00=00 =FF=
=DA=00=08=01=03=01=01?=10=1F=FF=C4=00=14=11=01=00=00=00=00=00=00=00=00=00=
=00=00=00=00=00=00 =FF=DA=00=08=01=02=01=01?=10=1F=FF=C4=00=19=10=00=03=01=
=01=01=00=00=00=00=00=00=00=00=00=00=00=01=11!=101Q=FF=DA=00=08=01=01=00=01=
?=10=0F=90{=9E=EA=02=A2%=18=E8w=05=9E=87=9F=FF=D9=

--MIXED2
Date: Thu, 21 Jun 2018 11:00:06 -0400
MIME-Version: 1.0
Content-Type: text/plain
Content-Disposition: inline
Content-Transfer-Encoding: 7bit

This is text part D

--MIXED2--

--ALTERNATIVE1
Date: Thu, 21 Jun 2018 11:00:06 -0400
MIME-Version: 1.0
Content-Type: multipart/related; boundary="RELATED1";
Content-Transfer-Encoding: 7bit

--RELATED1
Date: Thu, 21 Jun 2018 11:00:06 -0400
MIME-Version: 1.0
Content-Type: text/html
Content-Transfer-Encoding: 7bit

<html><body> This is html part E </body></html>

--RELATED1
Date: Thu, 21 Jun 2018 12:00:19 -0400
MIME-Version: 1.0
Content-Type: image/jpeg; name="f.jpg"
Content-Transfer-Encoding: quoted-printable
Content-Disposition: inline

=FF=D8=FF=E0=00=10JFIF=00=01=01=01=00H=00H=00=00=FF=FE=00=13Created with GI=
MP=FF=DB=00C=00=03=02=02=03=02=02=03=03=03=03=04=03=03=04=05=08=05=05=04=04=
=05
=07=07=06=08=0C
=0C=0C=0B
=0B=0B=0D=0E=12=10=0D=0E=11=0E=0B=0B=10=16=10=11=13=14=15=15=15=0C=0F=17=18=
=16=14=18=12=14=15=14=FF=DB=00C=01=03=04=04=05=04=05	=05=05	=14=0D=0B=0D=14=
=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=
=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=FF=
=C2=00=11=08=00
=00
=03=01=11=00=02=11=01=03=11=01=FF=C4=00=16=00=01=01=01=00=00=00=00=00=00=00=
=00=00=00=00=00=00=00=06=07=FF=C4=00=14=01=01=00=00=00=00=00=00=00=00=00=00=
=00=00=00=00=00=00=FF=DA=00=0C=03=01=00=02=10=03=10=00=00=01=DCK=80=0F=FF=
=C4=00=18=10=00=02=03=00=00=00=00=00=00=00=00=00=00=00=00=00=01=03=04=06=10=
=FF=DA=00=08=01=01=00=01=05=02-=9A=EBv=FF=00=FF=C4=00=14=11=01=00=00=00=00=
=00=00=00=00=00=00=00=00=00=00=00 =FF=DA=00=08=01=03=01=01?=01=1F=FF=C4=00=
=14=11=01=00=00=00=00=00=00=00=00=00=00=00=00=00=00=00 =FF=DA=00=08=01=02=
=01=01?=01=1F=FF=C4=00=1C=10=00=02=02=02=03=00=00=00=00=00=00=00=00=00=00=
=00=01=02=03=11=04 1QR=FF=DA=00=08=01=01=00=06?=020=89=99pa=C7I=0C`=A8=B6%=
=C7=9B<v+O=FF=C4=00=18=10=01=01=00=03=00=00=00=00=00=00=00=00=00=00=00=00=
=011=11 !=FF=DA=00=08=01=01=00=01?!=BA=E4=A6P(y=92\=E9=FF=DA=00=0C=03=01=00=
=02=00=03=00=00=00=10=12O=FF=C4=00=14=11=01=00=00=00=00=00=00=00=00=00=00=
=00=00=00=00=00 =FF=DA=00=08=01=03=01=01?=10=1F=FF=C4=00=14=11=01=00=00=00=
=00=00=00=00=00=00=00=00=00=00=00=00 =FF=DA=00=08=01=02=01=01?=10=1F=FF=C4=
=00=17=10=01=01=01=01=00=00=00=00=00=00=00=00=00=00=00=00=01=11=10Q=FF=DA=
=00=08=01=01=00=01?=10
=A2=ADJy=B2=B4)i=BF=FF=D9=

--RELATED1--

--ALTERNATIVE1--

--MIXED1
Date: Thu, 21 Jun 2018 12:00:19 -0400
MIME-Version: 1.0
Content-Type: image/jpeg; name="g.jpg"
Content-Disposition: attachment; filename="g.jpg"
Content-Transfer-Encoding: quoted-printable

=FF=D8=FF=E0=00=10JFIF=00=01=01=01=00H=00H=00=00=FF=FE=00=13Created with GI=
MP=FF=DB=00C=00=03=02=02=03=02=02=03=03=03=03=04=03=03=04=05=08=05=05=04=04=
=05
=07=07=06=08=0C
=0C=0C=0B
=0B=0B=0D=0E=12=10=0D=0E=11=0E=0B=0B=10=16=10=11=13=14=15=15=15=0C=0F=17=18=
=16=14=18=12=14=15=14=FF=DB=00C=01=03=04=04=05=04=05	=05=05	=14=0D=0B=0D=14=
=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=
=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=14=FF=
=C2=00=11=08=00
=00
=03=01=11=00=02=11=01=03=11=01=FF=C4=00=15=00=01=01=00=00=00=00=00=00=00=00=
=00=00=00=00=00=00=00=07=FF=C4=00=14=01=01=00=00=00=00=00=00=00=00=00=00=00=
=00=00=00=00=00=FF=DA=00=0C=03=01=00=02=10=03=10=00=00=01=A3=160=0F=FF=C4=
=00=17=10=01=00=03=00=00=00=00=00=00=00=00=00=00=00=00=00=04=05=10=15=FF=DA=
=00=08=01=01=00=01=05=02T=EA=B5o=FF=C4=00=14=11=01=00=00=00=00=00=00=00=00=
=00=00=00=00=00=00=00 =FF=DA=00=08=01=03=01=01?=01=1F=FF=C4=00=14=11=01=00=
=00=00=00=00=00=00=00=00=00=00=00=00=00=00 =FF=DA=00=08=01=02=01=01?=01=1F=
=FF=C4=00=1D=10=00=02=01=04=03=00=00=00=00=00=00=00=00=00=00=00=01=03=02=04=
=13!"=14 Q=FF=DA=00=08=01=01=00=06?=02b=F9nC=85U=95=D1=A9p=96=98=DE@=ECA=F6=
'=1D?=FF=C4=00=18=10=01=01=00=03=00=00=00=00=00=00=00=00=00=00=00=00=01=11=
=10!q=FF=DA=00=08=01=01=00=01?!CMK=FB=C1=E5=D5A=1CW?=FF=DA=00=0C=03=01=00=
=02=00=03=00=00=00=10=92O=FF=C4=00=14=11=01=00=00=00=00=00=00=00=00=00=00=
=00=00=00=00=00 =FF=DA=00=08=01=03=01=01?=10=1F=FF=C4=00=14=11=01=00=00=00=
=00=00=00=00=00=00=00=00=00=00=00=00 =FF=DA=00=08=01=02=01=01?=10=1F=FF=C4=
=00=17=10=01=01=01=01=00=00=00=00=00=00=00=00=00=00=00=00=01=11 A=FF=DA=00=
=08=01=01=00=01?=10=9D	=B7C"0"=9D=C4=7F=FF=D9=

--MIXED1
Date: Thu, 21 Jun 2018 11:00:06 -0400
MIME-Version: 1.0
Content-Type: application/x-excel
Content-Transfer-Encoding: 7bit

XXX Excelt H

--MIXED1
Date: Thu, 21 Jun 2018 11:00:06 -0400
MIME-Version: 1.0
Content-Type: message/rfc822
Content-Transfer-Encoding: 7bit

Date: Thu, 21 Jun 2018 11:00:06 -0400
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Content-Type: text/plain

RFC822 message

--MIXED1--

--TOP
Date: Thu, 21 Jun 2018 11:00:06 -0400
MIME-Version: 1.0
Content-Type: text/plain
Content-Disposition: inline
Content-Transfer-Encoding: 7bit

This is text part K

--TOP--
````

The import seems to go fine. Then when I request it back:

````
  my $res = $tester->request({
    using => [ "ietf:jmapmail" ],
    methodCalls => [[
      "Email/get" => {
        ids        => [ $message->id ],
        properties => [ 'textBody', 'bodyValues', 'bodyStructure' ],
        fetchAllBodyValues => jtrue(),
      },
    ]],
  });

  diag explain $res->as_stripped_triples;
````

I get this response:

````
        # [
        #   [
        #     'Email/get',
        #     {
        #       'accountId' => 'jt-588ba5d6-7595-11e8-9c0f-a005ea1562dd',
        #       'list' => [
        #         {
        #           'bodyStructure' => {
        #             'blobId' => undef,
        #             'charset' => undef,
        #             'cid' => undef,
        #             'disposition' => undef,
        #             'language' => [],
        #             'location' => undef,
        #             'name' => undef,
        #             'partId' => undef,
        #             'size' => 0,
        #             'subParts' => [
        #               {
        #                 'blobId' => 'Gca635541ee9a6f9498dd7623cb78bd668f3dd417',
        #                 'charset' => 'us-ascii',
        #                 'cid' => undef,
        #                 'disposition' => undef,
        #                 'language' => [],
        #                 'location' => undef,
        #                 'name' => undef,
        #                 'partId' => '1',
        #                 'size' => 21,
        #                 'type' => 'text/plain'
        #               },
        #               {
        #                 'blobId' => undef,
        #                 'charset' => undef,
        #                 'cid' => undef,
        #                 'disposition' => undef,
        #                 'language' => [],
        #                 'location' => undef,
        #                 'name' => undef,
        #                 'partId' => undef,
        #                 'size' => 0,
        #                 'subParts' => [
        #                   {
        #                     'blobId' => undef,
        #                     'charset' => undef,
        #                     'cid' => undef,
        #                     'disposition' => undef,
        #                     'language' => [],
        #                     'location' => undef,
        #                     'name' => undef,
        #                     'partId' => undef,
        #                     'size' => 0,
        #                     'subParts' => [
        #                       {
        #                         'blobId' => undef,
        #                         'charset' => undef,
        #                         'cid' => undef,
        #                         'disposition' => undef,
        #                         'language' => [],
        #                         'location' => undef,
        #                         'name' => undef,
        #                         'partId' => undef,
        #                         'size' => 0,
        #                         'subParts' => [
        #                           {
        #                             'blobId' => 'Gaf0399b679105a7b96f9a897b2c974b884d27afb',
        #                             'charset' => 'us-ascii',
        #                             'cid' => undef,
        #                             'disposition' => 'inline',
        #                             'language' => [],
        #                             'location' => undef,
        #                             'name' => undef,
        #                             'partId' => '2.1.1.1',
        #                             'size' => 21,
        #                             'type' => 'text/plain'
        #                           },
        #                           {
        #                             'blobId' => 'G1e36887f7fc9f4f030d1a574ba3c26f35bd134b1',
        #                             'charset' => undef,
        #                             'cid' => undef,
        #                             'disposition' => 'inline',
        #                             'language' => [],
        #                             'location' => undef,
        #                             'name' => 'c.jpg',
        #                             'partId' => undef,
        #                             'size' => 630,
        #                             'type' => 'image/jpeg'
        #                           },
        #                           {
        #                             'blobId' => 'Gdf3a02968c945bf674d180d76cbd39bc2a4a2b30',
        #                             'charset' => 'us-ascii',
        #                             'cid' => undef,
        #                             'disposition' => 'inline',
        #                             'language' => [],
        #                             'location' => undef,
        #                             'name' => undef,
        #                             'partId' => '2.1.1.3',
        #                             'size' => 21,
        #                             'type' => 'text/plain'
        #                           }
        #                         ],
        #                         'type' => 'multipart/mixed'
        #                       },
        #                       {
        #                         'blobId' => undef,
        #                         'charset' => undef,
        #                         'cid' => undef,
        #                         'disposition' => undef,
        #                         'language' => [],
        #                         'location' => undef,
        #                         'name' => undef,
        #                         'partId' => undef,
        #                         'size' => 0,
        #                         'subParts' => [
        #                           {
        #                             'blobId' => 'G45a24429dda6219724bde535c084d7009bd41506',
        #                             'charset' => 'us-ascii',
        #                             'cid' => undef,
        #                             'disposition' => undef,
        #                             'language' => [],
        #                             'location' => undef,
        #                             'name' => undef,
        #                             'partId' => '2.1.2.1',
        #                             'size' => 49,
        #                             'type' => 'text/html'
        #                           },
        #                           {
        #                             'blobId' => 'Gcbd0599bce687ef5c5d8945140bcba2e377b1ada',
        #                             'charset' => undef,
        #                             'cid' => undef,
        #                             'disposition' => 'inline',
        #                             'language' => [],
        #                             'location' => undef,
        #                             'name' => 'f.jpg',
        #                             'partId' => undef,
        #                             'size' => 608,
        #                             'type' => 'image/jpeg'
        #                           }
        #                         ],
        #                         'type' => 'multipart/related'
        #                       }
        #                     ],
        #                     'type' => 'multipart/alternative'
        #                   },
        #                   {
        #                     'blobId' => 'G3540041de8931997a2562291f681bf122cf56425',
        #                     'charset' => undef,
        #                     'cid' => undef,
        #                     'disposition' => 'attachment',
        #                     'language' => [],
        #                     'location' => undef,
        #                     'name' => 'g.jpg',
        #                     'partId' => undef,
        #                     'size' => 608,
        #                     'type' => 'image/jpeg'
        #                   },
        #                   {
        #                     'blobId' => 'G246b0ade30ee6247517b5ea27c79609e49a6bba1',
        #                     'charset' => undef,
        #                     'cid' => undef,
        #                     'disposition' => undef,
        #                     'language' => [],
        #                     'location' => undef,
        #                     'name' => undef,
        #                     'partId' => undef,
        #                     'size' => 14,
        #                     'type' => 'application/x-excel'
        #                   },
        #                   {
        #                     'blobId' => 'Gbc10021debd0c0eb0ecc502c21f091819269d09f',
        #                     'charset' => undef,
        #                     'cid' => undef,
        #                     'disposition' => undef,
        #                     'language' => [],
        #                     'location' => undef,
        #                     'name' => undef,
        #                     'partId' => undef,
        #                     'size' => 135,
        #                     'type' => 'message/rfc822'
        #                   }
        #                 ],
        #                 'type' => 'multipart/mixed'
        #               },
        #               {
        #                 'blobId' => 'Ga309fbbeef3984ef0b668ccbee5ba2457929e3d5',
        #                 'charset' => 'us-ascii',
        #                 'cid' => undef,
        #                 'disposition' => 'inline',
        #                 'language' => [],
        #                 'location' => undef,
        #                 'name' => undef,
        #                 'partId' => '3',
        #                 'size' => 21,
        #                 'type' => 'text/plain'
        #               }
        #             ],
        #             'type' => 'multipart/mixed'
        #           },
        #           'bodyValues' => {
        #             '1' => {
        #               'isEncodingProblem' => bless( do{\(my $o = 0)}, 'JSON::PP::Boolean' ),
        #               'isTruncated' => $VAR1->[0][1]{'list'}[0]{'bodyValues'}{'1'}{'isEncodingProblem'},
        #               'value' => 'This is text part A
        # '
        #             },
        #             '2.1.1.1' => {
        #               'isEncodingProblem' => $VAR1->[0][1]{'list'}[0]{'bodyValues'}{'1'}{'isEncodingProblem'},
        #               'isTruncated' => $VAR1->[0][1]{'list'}[0]{'bodyValues'}{'1'}{'isEncodingProblem'},
        #               'value' => 'This is text part B
        # '
        #             },
        #             '2.1.1.3' => {
        #               'isEncodingProblem' => $VAR1->[0][1]{'list'}[0]{'bodyValues'}{'1'}{'isEncodingProblem'},
        #               'isTruncated' => $VAR1->[0][1]{'list'}[0]{'bodyValues'}{'1'}{'isEncodingProblem'},
        #               'value' => 'This is text part D
        # '
        #             },
        #             '2.1.2.1' => {
        #               'isEncodingProblem' => $VAR1->[0][1]{'list'}[0]{'bodyValues'}{'1'}{'isEncodingProblem'},
        #               'isTruncated' => $VAR1->[0][1]{'list'}[0]{'bodyValues'}{'1'}{'isEncodingProblem'},
        #               'value' => '<html><body> This is html part E </body></html>
        # '
        #             },
        #             '3' => {
        #               'isEncodingProblem' => $VAR1->[0][1]{'list'}[0]{'bodyValues'}{'1'}{'isEncodingProblem'},
        #               'isTruncated' => $VAR1->[0][1]{'list'}[0]{'bodyValues'}{'1'}{'isEncodingProblem'},
        #               'value' => 'This is text part K
        # '
        #             }
        #           },
        #           'id' => 'Mec1d3c4ed1ad4f508ef76582',
        #           'textBody' => [
        #             {
        #               'blobId' => 'Gca635541ee9a6f9498dd7623cb78bd668f3dd417',
        #               'charset' => 'us-ascii',
        #               'cid' => undef,
        #               'disposition' => undef,
        #               'language' => [],
        #               'location' => undef,
        #               'name' => undef,
        #               'partId' => '1',
        #               'size' => 21,
        #               'type' => 'text/plain'
        #             },
        #             {
        #               'blobId' => 'Gaf0399b679105a7b96f9a897b2c974b884d27afb',
        #               'charset' => 'us-ascii',
        #               'cid' => undef,
        #               'disposition' => 'inline',
        #               'language' => [],
        #               'location' => undef,
        #               'name' => undef,
        #               'partId' => '2.1.1.1',
        #               'size' => 21,
        #               'type' => 'text/plain'
        #             },
        #             {
        #               'blobId' => 'G1e36887f7fc9f4f030d1a574ba3c26f35bd134b1',
        #               'charset' => undef,
        #               'cid' => undef,
        #               'disposition' => 'inline',
        #               'language' => [],
        #               'location' => undef,
        #               'name' => 'c.jpg',
        #               'partId' => undef,
        #               'size' => 630,
        #               'type' => 'image/jpeg'
        #             },
        #             {
        #               'blobId' => 'Gdf3a02968c945bf674d180d76cbd39bc2a4a2b30',
        #               'charset' => 'us-ascii',
        #               'cid' => undef,
        #               'disposition' => 'inline',
        #               'language' => [],
        #               'location' => undef,
        #               'name' => undef,
        #               'partId' => '2.1.1.3',
        #               'size' => 21,
        #               'type' => 'text/plain'
        #             },
        #             {
        #               'blobId' => 'Ga309fbbeef3984ef0b668ccbee5ba2457929e3d5',
        #               'charset' => 'us-ascii',
        #               'cid' => undef,
        #               'disposition' => 'inline',
        #               'language' => [],
        #               'location' => undef,
        #               'name' => undef,
        #               'partId' => '3',
        #               'size' => 21,
        #               'type' => 'text/plain'
        #             }
        #           ]
        #         }
        #       ],
        #       'notFound' => [],
        #       'state' => '24'
        #     },
        #     'e'
        #   ]
        # ]
````

The structure itself looks correct and the textBody only gives us the expected elements (from the example).

HOWEVER.

The image/*, message/* and application/* parts have no partId. From reading the spec, I think they *should* (but maybe it's not useful at all?):

````
partId: String|null Identifies this part uniquely within the Email. This is scoped to the emailId and has no meaning outside of the JMAP Email object representation. This is null if, and only if, the part is of type multipart/*.
````

**Comment by brong** [comment]
Yeah - it should be on everything except mulitpart:

Fixed in 8c121fc97f5fdee572f416d4eb1f7bdae3a7cc89
